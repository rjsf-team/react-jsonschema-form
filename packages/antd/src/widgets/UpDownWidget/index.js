import React from 'react';

import InputNumber from 'antd/lib/input-number';

const INPUT_STYLE = {
  width: '100%',
};

const UpDownWidget = ({
  // autofocus,
  disabled,
  formContext,
  id,
  onBlur,
  onChange,
  onFocus,
  // options,
  placeholder,
  readonly,
  // required,
  // schema,
  value,
}) => {
  const { readonlyAsDisabled = true } = formContext;

  const handleChange = (nextValue) => onChange(nextValue);

  const handleBlur = ({ target }) => onBlur(id, target.value);

  const handleFocus = ({ target }) => onFocus(id, target.value);

  return (
    <InputNumber
      disabled={disabled || (readonlyAsDisabled && readonly)}
      id={id}
      name={id}
      onBlur={!readonly ? handleBlur : undefined}
      onChange={!readonly ? handleChange : undefined}
      onFocus={!readonly ? handleFocus : undefined}
      placeholder={placeholder}
      style={INPUT_STYLE}
      type="number"
      value={value}
    />
  );
};

export default UpDownWidget;
