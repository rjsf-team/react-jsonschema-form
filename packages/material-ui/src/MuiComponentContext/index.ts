export { default } from './MuiComponentContext';
export * from './MuiComponentContext';
export * from './useMuiComponent';
