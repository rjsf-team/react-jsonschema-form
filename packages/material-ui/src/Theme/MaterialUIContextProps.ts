import { ComponentType } from 'react';
import type {
  BoxProps,
  ButtonProps,
  CheckboxProps,
  DividerProps,
  GridProps,
  FormControlProps,
  FormControlLabelProps,
  FormGroupProps,
  FormHelperTextProps,
  FormLabelProps,
  IconButtonProps,
  InputProps,
  InputLabelProps,
  ListProps,
  ListItemProps,
  ListItemIconProps,
  ListItemTextProps,
  MenuItemProps,
  PaperProps,
  RadioProps,
  RadioGroupProps,
  SliderProps,
  SvgIconProps,
  TextFieldProps,
  TypographyProps,
} from '@material-ui/core';

export default interface MaterialUIContextProps {
  Box: ComponentType<BoxProps>;
  Button: ComponentType<ButtonProps>;
  Checkbox: ComponentType<CheckboxProps>;
  Divider: ComponentType<DividerProps>;
  Grid: ComponentType<GridProps>;
  FormControl: ComponentType<FormControlProps>;
  FormControlLabel: ComponentType<FormControlLabelProps>;
  FormGroup: ComponentType<FormGroupProps>;
  FormHelperText: ComponentType<FormHelperTextProps>;
  FormLabel: ComponentType<FormLabelProps>;
  IconButton: ComponentType<IconButtonProps>;
  Input: ComponentType<InputProps>;
  InputLabel: ComponentType<InputLabelProps>;
  List: ComponentType<ListProps>;
  ListItem:ComponentType<ListItemProps>;
  ListItemIcon: ComponentType<ListItemIconProps>;
  ListItemText: ComponentType<ListItemTextProps>;
  MenuItem: ComponentType<MenuItemProps>;
  Paper: ComponentType<PaperProps>;
  Radio: ComponentType<RadioProps>;
  RadioGroup: ComponentType<RadioGroupProps>;
  Slider: ComponentType<SliderProps>;
  TextField: ComponentType<Omit<TextFieldProps, 'color' | 'variant'>>;
  Typography: ComponentType<TypographyProps>;
  AddIcon: ComponentType<SvgIconProps>;
  ArrowDownwardIcon: ComponentType<SvgIconProps>;
  ArrowUpwardIcon: ComponentType<SvgIconProps>;
  ErrorIcon: ComponentType<SvgIconProps>;
  RemoveIcon: ComponentType<SvgIconProps>;
}
