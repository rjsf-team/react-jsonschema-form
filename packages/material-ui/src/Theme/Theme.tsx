import React from "react";

import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";

import ArrayFieldTemplate from "../ArrayFieldTemplate";
import ErrorList from "../ErrorList";
import Fields from "../Fields";
import FieldTemplate from "../FieldTemplate";
import ObjectFieldTemplate from "../ObjectFieldTemplate";
import Widgets from "../Widgets";

import { ThemeProps } from "@rjsf/core";
import { utils } from "@rjsf/core";
const { getDefaultRegistry } = utils;

const { fields, widgets } = getDefaultRegistry();

const DefaultChildren = () => (
  <Box marginTop={3}>
    <Button type="submit" variant="contained" color="primary">
      Submit
    </Button>
  </Box>
);

const Theme: ThemeProps = {
  children: <DefaultChildren />,
  ArrayFieldTemplate,
  fields: { ...fields, ...Fields },
  FieldTemplate,
  ObjectFieldTemplate,
  widgets: { ...widgets, ...Widgets },
  ErrorList,
};

export default Theme;
