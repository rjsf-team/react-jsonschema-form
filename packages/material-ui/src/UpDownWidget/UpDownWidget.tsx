import React from 'react';

import FormControl from '@material-ui/core/FormControl';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';

import { WidgetProps } from '@rjsf/core';

const UpDownWidget = ({
  id,
  required,
  readonly,
  disabled,
  label,
  value,
  onChange,
  onBlur,
  onFocus,
  autofocus,
}: WidgetProps) => {
  const _onChange = ({
    target: { value },
  }: React.ChangeEvent<HTMLInputElement>) => onChange(value);
  const _onBlur = ({ target: { value } }: React.FocusEvent<HTMLInputElement>) =>
    onBlur(id, value);
  const _onFocus = ({
    target: { value },
  }: React.FocusEvent<HTMLInputElement>) => onFocus(id, value);

  return (
    <FormControl
      fullWidth={true}
      //error={!!rawErrors}
      required={required}
    >
      <InputLabel>{label}</InputLabel>
      <Input
        id={id}
        autoFocus={autofocus}
        required={required}
        type="number"
        disabled={disabled || readonly}
        value={value || value === 0 ? value : ''}
        onChange={_onChange}
        onBlur={_onBlur}
        onFocus={_onFocus}
      />
    </FormControl>
  );
};

export default UpDownWidget;
