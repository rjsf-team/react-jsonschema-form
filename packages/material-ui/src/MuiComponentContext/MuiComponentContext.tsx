import React from 'react';
import MaterialUIContextProps from '../Theme/MaterialUIContextProps';
import Mui5ContextProps from '../Theme5/Mui5ContextProps';

/** This context is used to encapsulate the usage of `@material-ui` or `@mui` away from the field, widgets and helpers.
 * It can be created using either an instance of `MaterialUIContextProps` or `Mui5ContextProps`.
 */
export default React.createContext<MaterialUIContextProps | Mui5ContextProps | null>(null);
