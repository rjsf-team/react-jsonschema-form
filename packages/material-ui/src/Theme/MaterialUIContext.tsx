import React, { ComponentType } from 'react';
import { BoxProps, ButtonProps } from '@material-ui/core';
import MaterialUIContextProps, { defaultMaterialUIContext } from './MaterialUIContextProps';

export const MaterialUIContext: MaterialUIContextProps = defaultMaterialUIContext;

try {
  /**
   * Dynamically import MUI components so we get full tree-shaking support
   */
  import('@material-ui/icons/Add').then(Add => {
    MaterialUIContext.AddIcon = Add.default;
  });
  import('@material-ui/icons/ArrowDownward').then(ArrowDownward => {
    MaterialUIContext.ArrowDownwardIcon = ArrowDownward.default;
  });
  import('@material-ui/icons/ArrowUpward').then(ArrowUpward => {
    MaterialUIContext.ArrowUpwardIcon = ArrowUpward.default;
  });
  import('@material-ui/core/Box').then(Box => {
    MaterialUIContext.Box = Box.default;
  });
  import('@material-ui/core/Button').then(Button => {
    MaterialUIContext.Button = Button.default;
  });
  import('@material-ui/core/Checkbox').then(Checkbox => {
    MaterialUIContext.Checkbox = Checkbox.default;
  });
  import('@material-ui/core/Divider').then(Divider => {
    MaterialUIContext.Divider = Divider.default;
  });
  import('@material-ui/icons/Error').then(Error => {
    MaterialUIContext.ErrorIcon = Error.default;
  });
  import('@material-ui/core/FormControl').then(FormControl => {
    MaterialUIContext.FormControl = FormControl.default;
  });
  import('@material-ui/core/FormControlLabel').then(FormControlLabel => {
    MaterialUIContext.FormControlLabel = FormControlLabel.default;
  });
  import('@material-ui/core/FormGroup').then(FormGroup => {
    MaterialUIContext.FormGroup = FormGroup.default;
  });
  import('@material-ui/core/FormHelperText').then(FormHelperText => {
    MaterialUIContext.FormHelperText = FormHelperText.default;
  });
  import('@material-ui/core/FormLabel').then(FormLabel => {
    MaterialUIContext.FormLabel = FormLabel.default;
  });
  import('@material-ui/core/Grid').then(Grid => {
    MaterialUIContext.Grid = Grid.default;
  });
  import('@material-ui/core/IconButton').then(IconButton => {
    MaterialUIContext.IconButton = IconButton.default;
  });
  import('@material-ui/core/OutlinedInput').then(OutlinedInput => {
    MaterialUIContext.Input = OutlinedInput.default;
  });
  import('@material-ui/core/InputLabel').then(InputLabel => {
    MaterialUIContext.InputLabel = InputLabel.default;
  });
  import('@material-ui/core/List').then(List => {
    MaterialUIContext.List = List.default;
  });
  import('@material-ui/core/ListItem').then(ListItem => {
    MaterialUIContext.ListItem = ListItem.default as ComponentType;
  });
  import('@material-ui/core/ListItemIcon').then(ListItemIcon => {
    MaterialUIContext.ListItemIcon = ListItemIcon.default;
  });
  import('@material-ui/core/ListItemText').then(ListItemText => {
    MaterialUIContext.ListItemText = ListItemText.default;
  });
  import('@material-ui/core/MenuItem').then(MenuItem => {
    MaterialUIContext.MenuItem = MenuItem.default as ComponentType;
  });
  import('@material-ui/core/Paper').then(Paper => {
    MaterialUIContext.Paper = Paper.default;
  });
  import('@material-ui/core/Radio').then(Radio => {
    MaterialUIContext.Radio = Radio.default;
  });
  import('@material-ui/core/RadioGroup').then(RadioGroup => {
    MaterialUIContext.RadioGroup = RadioGroup.default;
  });
  import('@material-ui/icons/Remove').then(Remove => {
    MaterialUIContext.RemoveIcon = Remove.default;
  });
  import('@material-ui/core/Slider').then(Slider => {
    MaterialUIContext.Slider = Slider.default;
  });
  import('@material-ui/core/SvgIcon').then(SvgIcon => {
    MaterialUIContext.SvgIcon = SvgIcon.default;
  });
  import('@material-ui/core/TextField').then(TextField => {
    MaterialUIContext.TextField = TextField.default;
  });
  import('@material-ui/core/Typography').then(Typography => {
    MaterialUIContext.Typography = Typography.default;
  });
} catch (err) {
  // no-op
}

export const DefaultChildren = () => {
  if (
    Object.keys(MaterialUIContext).length &&
    MaterialUIContext.Box &&
    MaterialUIContext.Button &&
    MaterialUIContext.AddIcon
  ) {
    const { Box, Button } = MaterialUIContext;
    const BoxComp = Box as ComponentType<BoxProps>;
    const ButtonComp = Button as ComponentType<ButtonProps>;

    return (
      <BoxComp marginTop={3}>
        <ButtonComp type='submit' variant='contained' color='primary'>
          Submit
        </ButtonComp>
      </BoxComp>
    );
  }

  return <div>@material-ui not available</div>;
};

export function useMaterialUIContext(): MaterialUIContextProps {
  return MaterialUIContext;
}
