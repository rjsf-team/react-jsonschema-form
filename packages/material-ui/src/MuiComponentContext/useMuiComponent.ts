import { useContext } from 'react';
import MaterialUIContextProps from '../Theme/MaterialUIContextProps';
import Mui5ContextProps from '../Theme5/Mui5ContextProps';
import MuiComponentContext from './MuiComponentContext';

export function useMuiComponent(): MaterialUIContextProps | Mui5ContextProps {
  const muiComponents = useContext(MuiComponentContext);

  if (!muiComponents) {
    throw new Error(
      'Either v4 or v5 of material-ui components and icons must be installed as dependencies'
    );
  }

  return muiComponents;
}
