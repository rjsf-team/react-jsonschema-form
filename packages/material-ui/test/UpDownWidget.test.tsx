import React from "react";
import { JSONSchema7 } from "json-schema";
import renderer from "react-test-renderer";
import { WidgetProps } from "@rjsf/core";

import MuiComponentContext from '../src/MuiComponentContext/MuiComponentContext';
import { MaterialUIContext } from '../src/Theme/MaterialUIContext';
import UpDownWidget from "../src/UpDownWidget/UpDownWidget";

export const mockSchema: JSONSchema7 = {
  type: "array",
  items: {
    type: "string",
  },
};

export const mockEventHandlers = (): void => void 0;

test("renders 0 as 0 and not ''", () => {
  const props: WidgetProps = {
    uiSchema: {},
    schema: mockSchema,
    required: true,
    disabled: false,
    readonly: true,
    autofocus: true,
    label: "Some simple label",
    onChange: mockEventHandlers,
    onBlur: mockEventHandlers,
    onFocus: mockEventHandlers,
    multiple: false,
    rawErrors: [""],
    options: {},
    formContext: {},
    id: "_id",
    placeholder: "",
    value: 0,
    registry: {
      fields: {},
      widgets: {},
      definitions: {},
      formContext: {},
      rootSchema: {},
    }
  };
  const tree = renderer.create((
    <MuiComponentContext.Provider value={MaterialUIContext}>
      <UpDownWidget {...props} />
    </MuiComponentContext.Provider>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
describe("UpDownWidget", () => {
});
