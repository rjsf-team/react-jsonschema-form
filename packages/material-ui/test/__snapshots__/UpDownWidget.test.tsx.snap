// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 0 as 0 and not '' 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
    data-shrink={true}
  >
    Some simple label
    <span
      aria-hidden={true}
      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
    >
      â€‰
      *
    </span>
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-formControl"
    onClick={[Function]}
    required={true}
  >
    <input
      aria-invalid={false}
      autoFocus={true}
      className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
      disabled={true}
      id="_id"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="number"
      value={0}
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-2"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
